### DISCUSSION TITLE
JSONata - Filter for multiple values

### DISCUSSION THREAD

**Discussion by nileger** [grafana/grafana-infinity-datasource#385] [Q&A] [answered]
In Grafana, I use a dashboard variable which I use to filter my JSON data:
```
| jsonata "$[`ENVIRONMENT`=\"${environment}\"]"
```
Works perfectly!

Beyond that, I also want to allow users to select all environments. I tried to insert `PROD|INT|DEV` (instead of  `${environment}`) in the above query, which didn't work. It seems like the `|` isn't considered a logical operator but is included in the string.

How could I adapt the above query to allow the selection of multiple values?

Many thanks in advance!

**Comment by yesoreyeram** [comment, marked as answer]
I think you need to refer https://github.com/yesoreyeram/grafana-infinity-datasource/discussions/329#discussioncomment-2967372

**Reply by nileger** [reply]
Perfect, many thanks! The entire time I tried to use a regex with `~>` . Now, I ended up with the following:
```
| jsonata "$[`ENVIRONMENT` in [${environment:singlequote}]]"
```
